### DISCUSSION TITLE
qdmr crash during radio read

### DISCUSSION THREAD

**Discussion by JebBaker** [hmatuschek/qdmr#208] [Q&A]
As title states the program crashes during/after radio download/read

Updated Raspberry OS.  
TYT-MD-2017
I tried looking at Syslog during the process and the only log entries where where the radio was connected/disconnected.  
I looked for logs elsewhere but I am not finding them.  I am linux basic and would think that there must be a log somewhere.  

De KG5YUN

**Comment by hmatuschek** [comment]
You can get some log messages by starting qdmr from the command line. 

However, there is a better way. Usually qdmr cashes due to a bug in the decoding of the codeplug and not necessarily during the read itself. You can split this read/decode operation using the command line. There is a command line tool call `dmrconf ` that uses the same library as qdmr.

To read the binary codeplug from the device simply run
```
dmrconf read --verbose codeplug_md2017.dfu
``` 
This creates a file called `codeplug_md2017.dfu` containing the binary codeplug. You can then decode the codeplug to a human-readable YAML file by running
```
dmrconf decode --verbose --radio=md2017 codeplug_md2017.dfu codeplug.yaml
```
If the second step fails, the issue is with the decoding of the codeplug. Could you then send me the binary one, so I can reproduce the issue?

**Reply by JebBaker** [reply]
Here is the out put of the above command
dmrconf read --verbose codeplug_md2017.dfu
Debug in dfu_libusb.cc@39: Try to detect USB DFU interface 483:df11.
Debug in dfu_libusb.cc@72: Connected to DFU device 483:df11.
[                                                  ] 0%
[================================================= ] 99%
ERROR in dfu_libusb.cc@143: Cannot get status: Pipe error.
Debug in readcodeplug.cc@62: Save codeplug at 'codeplug_md2017.dfu'.


and 
dmrconf decode --verbose --radio=md2017 codeplug_md2017.dfu codeplug_1.yaml
Debug in radioid.cc@158: Automatically set default radio id to WA9JG.
Debug in tyt_codeplug.cc@1545: Explicitly set default ID to index=0.
Segmentation fault

Thats what I have. 

[================================================= ] 99%
ERROR in dfu_libusb.cc@143: Cannot get status: Pipe error.
Debug in readcodeplug.cc@62: Save codeplug at 'codeplug_md2017.dfu'.

**Reply by hmatuschek** [reply]
Could you get me the binary codeplug? If there is nothing secret in the codeplug, you can attach it here. Otherwise just send it by email.

**Comment by JebBaker** [comment]
Thank you sir.

I will get the dmrconf tool installed and run the tasks you laid out and get you the results.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by JebBaker** [comment]
Oh yeah. For sure. It'll be a couple days.

We are going hamfest. I'll send it when I get back.

My name is Jeb. I live in Texas. This is really awesome.

Thank you for the work you gave done already.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by JebBaker** [comment]
here you go. 
[codeplug_md2017.zip](https://github.com/hmatuschek/qdmr/files/8364161/codeplug_md2017.zip)

**Comment by hmatuschek** [comment]
Ok, thanks. The funny thing is, that I can decode it using the current version 0.10.2. However, there are some warning concerning the scanlist. To this end, there might be an issue with the pirmary channel for the scan lists. I'll check that.

**Reply by hmatuschek** [reply]
There was in fact an issue. However, it should not cause a crash. This is fixed now in 5135e40f8fc020647c4e68ff6cf54d3cd0219ad1.

**Comment by JebBaker** [comment]
Ok.  Great.  I am rather new to Raspberry/Linux.  Will I need to re-install from scratch?

**Reply by allesand** [reply]
No re-install needed. http://archive.raspbian.org/raspbian/pool/main/q/qdmr/ has a recent version, so an "sudo apt update && sudo apt upgrade" should update the package list and install an updated version of qdmr.
